pkg/scheduler/core/spreadconstraint/group_clusters.go

```diff
@@ -128,6 +130,62 @@ func groupClustersIgnoringTopology(
 	return groupClustersInfo
 }
 
+// weightUnit is used to minimize the impact of avg(cluster.Score).
+// By multiply the weightUnit, the unit of targetReplica will be 1000, and the avg(cluster.Score) will in [0, 100].
+// Thus, when sorting by Group Score, targetReplica will be considered first, and if the Weights are the same, then Score will be considered.
+const weightUnit int64 = 1000
+
+func (info *GroupClustersInfo) calcGroupScore(
+	clusters []ClusterDetailInfo,
+	rbSpec *workv1alpha2.ResourceBindingSpec,
+	minGroups int) int64 {
+	// if the replica scheduling type is divided, the score is calculated by followed.
+	float64MinGroups := float64(minGroups)
+	targetReplica := int64(math.Ceil(float64(rbSpec.Replicas) / float64MinGroups))
+
+	// get the minGroups of Cluster.
+	var clusterMinGroups int
+	if rbSpec.Placement != nil && rbSpec.Placement.SpreadConstraints != nil {
+		for _, sc := range rbSpec.Placement.SpreadConstraints {
+			if sc.SpreadByField == policyv1alpha1.SpreadByFieldCluster {
+				clusterMinGroups = sc.MinGroups
+			}
+		}
+	}
+
+	// if the minGroups of Cluster is less than the minGroups of Group, set the minGroups of Cluster to the minGroups of Group.
+	if clusterMinGroups < minGroups {
+		clusterMinGroups = minGroups
+	}
+	int64ClusterMinGroups := int64(clusterMinGroups)
+
+	// clusters have been sorted by cluster.Score descending,
+	// and if the cluster.Score is the same, the cluster.availableReplica is ascending.
+	var sumAvailableReplica int64
+	var sumScore int64
+	var validClusters int64
+	for _, cluster := range clusters {
+		sumAvailableReplica += cluster.AvailableReplicas
+		sumScore += cluster.Score
+		validClusters++
+		if validClusters >= int64ClusterMinGroups && sumAvailableReplica >= targetReplica {
+			break
+		}
+	}
+
+	// cluster.Score is 0 or 100. To minimize the impact of Score,
+	// set the atomic value of targetReplica to 1000. This way,
+	// when sorting by Group Score, targetReplica will be considered first,
+	// and if the Weights are the same, then Score will be considered.
+	if sumAvailableReplica < targetReplica {
+		sumAvailableReplica = sumAvailableReplica * weightUnit
+		return sumAvailableReplica + sumScore/int64(len(clusters))
+	}
+
+	targetReplica = targetReplica * weightUnit
+	return targetReplica + sumScore/validClusters
```

Comment:
This piece of code is difficult to understand. Can we add some examples?  like https://github.com/karmada-io/karmada/blob/adef1e59748e1e1d31cb75fffe406b5dd69a66d7/pkg/scheduler/core/spreadconstraint/select_clusters_by_cluster.go#L57-L65

Reply:
Great, I’ll add the relevant comments. @RainbowMango